apps/backend/src/event/event.service.ts

```diff
@@ -26,11 +29,16 @@ export class EventService {
     }
   }
 
-  async update(id: string, data: Prisma.EventUpdateInput): Promise<Event> {
+  async update(id: string, data: UpdateEventDto): Promise<Event> {
     try {
       return await this.prisma.event.update({ where: { id }, data });
-    } catch {
-      throw new NotFoundException(`Event not found`);
+    } catch (error) {
+      if (error instanceof NotFoundException) {
```

Comment:
I don't think it will work like this, we should use the `PrismaClientKnownRequestError` since it contains the code of the error, from which we can decide what kind of error to throw.  Check the error codes in the Prisma docs (I already sent some links)
Here is an example from Konzisite:
![image](https://github.com/user-attachments/assets/494aef4f-eaed-4e94-87fb-528ca65dacc5)

Reply:
Just to make sure: P2001 should be the code if there's no event with the id, and P2006 should be the code if the data is invalid, right?
I'm not sure whether it is P2006 or P2007, but this looks more logical

Reply:
Hmm not sure, but please try out both cases and if you check the error code, you will see what kind of errors we get.